### PR TITLE
add test for #12009

### PR DESCRIPTION
**Pull request by KristofferC**

### CODE REVIEW

File: test/iterators.jl [hunk 1]
```diff
@@ -385,3 +385,7 @@ for T in (UInt8, UInt16, UInt32, UInt64, UInt128, Int8, Int16, Int128, BigInt)
     @test length(repeated(1, T(5))) == 5
     @test collect(partition(1:5, T(5)))[1] == collect(1:5)
 end
+
+@testset begin "collect finite iterators"
+    @test_nowarn collect(enumerate(Iterators.Filter(x -> x>0, randn(10))))
```
Comment (tkelman) [inline test/iterators.jl]: compare the return value against what it should be. and put the issue number in the testset description

Comment (KristofferC) [inline test/iterators.jl]: updated

**Comment by bramtayl** [comment]
Shouldn't it be

```julia
@testset "collect finite iterators" begin
```

instead? With the change in #20198, "collect finite iteratos" is getting parsed as a docstring.

**Comment by tkelman** [comment]
Yes, good catch

**Pull request merged**